### PR TITLE
1 Enhancement -  Network Compliance Module

### PR DESCRIPTION
**Pull request by rukapse**
**PR Summary:**

- Code changes have been made only to the plugins/modules/network_compliance_workflow_manager.py file. 
- Fixes Include:
   1.    Enhancement: Changed function="get_compliance_detail" to function="compliance_details_of_device"
- Tested on Catalyst Centre v2.3.7.0

Note: 
- This PR also includes a code change from previous PR, rectified a Log Message.

### CODE REVIEW

File: plugins/modules/network_compliance_workflow_manager.py [hunk 1]
```diff
@@ -874,6 +876,107 @@ def get_want(self, config):
...
+        """
+
+        # Initialize the lists/dicts
+        final_response = {}
+        device_list = []
+        compliance_details_of_device_params = {}
+
+        # Iterate through each device UUID in the run compliance parameters
+        for device_uuid in run_compliance_params["deviceUuids"]:
+
```
Comment (madhansansel) [inline plugins/modules/network_compliance_workflow_manager.py:903]: init
` device_ip = None`

File: plugins/modules/network_compliance_workflow_manager.py [hunk 2]
```diff
@@ -874,6 +876,107 @@ def get_want(self, config):
...
+            # Find the corresponding device IP for the given device UUID
+            for ip, device_id in mgmt_ip_instance_id_map.items():
+                if device_uuid == device_id:
+                    device_ip = ip
+                    break
+
```
Comment (madhansansel) [inline plugins/modules/network_compliance_workflow_manager.py:909]: What happens if device_ip is none? 

```
        if device_ip is None:
            log a warning message -- Device UUID {device_uuid} not found in mgmt_ip_instance_id_map" 
            continue
```

File: plugins/modules/network_compliance_workflow_manager.py [hunk 3]
```diff
@@ -874,6 +876,107 @@ def get_want(self, config):
...
+            # Initialize the response list for the device IP if not already present
+            if device_ip not in final_response.keys():
+                final_response[device_ip] = []
+
+            # Check if categories are specified and fetch details for each category of the device
+            if "categories" in run_compliance_params.keys():
+                for category in run_compliance_params["categories"]:
+                    compliance_details_of_device_params["category"] = category
+                    compliance_details_of_device_params["device_uuid"] = device_uuid
```
Comment (madhansansel) [inline plugins/modules/network_compliance_workflow_manager.py:921]: do we set device_uuid, and all inside for loop? Is it common for all the category?

Comment (rukapse) [reply plugins/modules/network_compliance_workflow_manager.py:921]: yes, so the new API that we are using is such that:
for a single device we can get all categories at once. But if we want category specific report, we have to do a API call for each category separately.

File: plugins/modules/network_compliance_workflow_manager.py [hunk 4]
```diff
@@ -874,6 +876,112 @@ def get_want(self, config):
...
+        for device_uuid in run_compliance_params["deviceUuids"]:
+
+            # Find the corresponding device IP for the given device UUID
+            for ip, device_id in mgmt_ip_instance_id_map.items():
+                if device_uuid == device_id:
+                    device_ip = ip
+                    break
+
+            if device_ip is None:
+                self.log("Device UUID: {0} not found in mgmt_ip_instance_id_map: {1}".format(device_uuid, mgmt_ip_instance_id_map), "ERROR")
```
[the remark this answers is not in the document]
Comment (rukapse) [reply plugins/modules/network_compliance_workflow_manager.py]: Fixed!

**Review by madhansansel — APPROVED** [review]

**Pull request merged**